src/Dotnet.Script.Core/NuGet/NuGetMetadataReferenceResolver.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+
+namespace Dotnet.Script.Core.NuGet
+{
+    /// <summary>
+    /// A <see cref="MetadataReferenceResolver"/> decorator that handles
+    /// references to NuGet packages in scripts.  
+    /// </summary>
+    public class NuGetMetadataReferenceResolver : MetadataReferenceResolver
```

Comment:
I've took most of the code from Dotnet.Script.NuGetMetadataResolver and tweaked it to have it work with csproj based scripting. I didn't want to mess up anything for OmniSharp which uses this package, so I figured we would benefit from having this code internally for now as we figure out all the details. When we have all the stuff working for csproj, we should probably move this out into NuGet packages again so that it can be consumed by for instance OmniSharp. When that time comes, the Dotnet.Script.NuGetMetadataResolver should be as small as possible containing nothing but the actual NuGetMetadataReferenceResolver class since that implementation basically just returns a dummy reference to itself.

The following code is from Dotnet.Script.NuGetMetadataResolver

```c#
if (reference.StartsWith("nuget", StringComparison.OrdinalIgnoreCase))
            {
                // HACK We need to return something here to "mark" the reference as resolved. 
                // https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Portable/ReferenceManager/CommonReferenceManager.Resolution.cs#L838
                return ImmutableArray<PortableExecutableReference>.Empty.Add(
                    MetadataReference.CreateFromFile(typeof(NuGetMetadataReferenceResolver).GetTypeInfo().Assembly.Location));
            }
```

You might ask why we don't simply return a assembly reference to say typeof(string)? 
The reason for that is we need to return something that matches the scripts target framework seen from OmniSharp.
When OmniSharp provides metadata references for a script, it does so in the context of the full framework unless the script has the following "shebang"

```c#
#! "netcoreapp1.1"
```

When that is missing, OmniSharp will provide references for the full fx and the we can not simply return a reference to typeof(string).Assembly since that would refer to the wrong "mscorlib".

The drawback of this approach is that we basically provide metadata for the whole Dotnet.Script.NuGetMetadataResolver assembly as a dummy reference. 

Back to the scripting side of things, we are always a netcore app and we can safely just return the assembly containg the string type

```c#
/ HACK We need to return something here to "mark" the reference as resolved. 
                // https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Portable/ReferenceManager/CommonReferenceManager.Resolution.cs#L838
                return ImmutableArray<PortableExecutableReference>.Empty.Add(
                    MetadataReference.CreateFromFile(typeof(string).GetTypeInfo().Assembly.Location));
```

Believe me when I say that this is the result of literally days of research and debugging :)

So until we figure out what can be shared between OmniSharp and script runners, I think we should just keep this internally for now as it allows us to move much faster. Hope this at least makes a little sense :)